pkg/kubelet/kubelet_node_status.go

```diff
@@ -265,7 +265,7 @@ func (kl *Kubelet) initialNode() (*v1.Node, error) {
 		node.Spec.Taints = nodeTaints
 	}
 	// Initially, set NodeNetworkUnavailable to true.
-	if kl.providerRequiresNetworkingConfiguration() {
+	if kl.providerRequiresNetworkingConfiguration() || kl.externalCloudProvider {
```

Comment:
This requires an update for the node controller, to remove the network unavailable node condition as well. Right now it only updates for the GCE provider for backwards compatibility reasons (see https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/cloud/node_controller.go#L242-L252)

Reply:
We need to be careful here because if nodes are upgraded with this change but the controllers are not, then you have nodes that will be stuck with this condition

Reply:
In addition, not all external cloud providers implements routes so adding this may not make sense as a catch all. Thoughts?